gaiku-common/src/data/chunk.rs

```diff
@@ -1,97 +1,125 @@
 use mint::{Vector3, Vector4};
 
-// TODO: Get inspiration on multiarray crate (https://github.com/sellibitze/multiarray) to make chunk 2d and 3d friendly
+pub trait Chunkify {
+  fn new(position: [f32; 3], width: usize, height: usize, depth: usize) -> Self;
+  fn depth(&self) -> usize;
+  fn is_air(&self, x: usize, y: usize, z: usize) -> bool;
+  fn get(&self, x: usize, y: usize, z: usize) -> u8;
```

Comment:
@QuantumEntangledAndy also for your use case to work it's ok to use `usize` or you need negative values too?

Reply:
Chunk space is always in uints no need for negative. What I do do is this:

```rust
// Populate the chunk
for chunk_x in 0..goxel_per_chunk[0] {
    let real_x = origin[0] + (chunk_x as f32 * actual_meter_per_goxel[0]);
    for chunk_z in 0..goxel_per_chunk[2] {
        let real_z = origin[2] + (chunk_z as f32 * actual_meter_per_goxel[2]);
        let value = noise.get([real_x as f64, real_z as f64]);
        for chunk_y in 0..goxel_per_chunk[1] {
            let real_y = origin[1] + (chunk_y as f32 * actual_meter_per_goxel[1]);
            if real_y as f64 <= value {
                chunk.set(chunk_x, chunk_y, chunk_z, 1);
                chunk.set_color(chunk_x, chunk_y, chunk_z, [0, 255, 255, 255].into());
            }
        }
    }
}
```

So I iter over chunk space. Convert chunk space to real space with an origin and voxels per meter. I also use differnt LODs so closer terrain has more voxels per meter but that's handled elsewhere.

Reply:
Do you think there might be a need to get `is_air` for points outside the chunk (at `-1` and `size+1`)? Maybe from neighbours.

Reply:
I was considering that scenario, for voxels is not that important, but for marching cubes this is a must have to calculate correctly the normals. The `update_neighbor_data` was where the data will be updated, but that will work only if you have a manager (I was thinking in an octree) where it call this method in each neighbor chunk when inserted/updated.

Other idea that pop out just recently was to add another param to the backer where you explicitly pass some options that could contain neighbor chunks, this will be easier to implement and but leave more work to the userland (that maybe is the best option).

But still I'm not sure how I want to solve this issue yet.

Reply:
In not sure about passing a whole neighbour chunk link in.

In my scenario those chunks change as the LOD changes so can't be sure it will survive for long and there won't be a chunk loaded for the end of the visibility range until the player approaches it.

I also load chunks asynchronously on multiple  thread so I can't pass multiple chunks to the baker since there are on different threads.

What I do have however is a function that can always tell you if it's air or not given a LOD resolution and a position which each chunk would already have access too.

Reply:
I will merge this PR and let's address your use case in another PR